soc/riscv/esp32c3/pinctrl_soc.h

```diff
@@ -32,34 +32,33 @@ typedef struct pinctrl_soc_pin {
  *
  * @param node_id Node identifier.
  */
-#define Z_PINCTRL_ESP32_PINMUX_INIT(node_id) DT_PROP(node_id, pinmux)
+#define Z_PINCTRL_ESP32_PINMUX_INIT(group_id, prop, idx) \
+	DT_PROP_BY_IDX(group_id, prop, idx)
 
 /**
  * @brief Utility macro to initialize pincfg field in #pinctrl_pin_t.
  *
  * @param node_id Node identifier.
  */
-#define Z_PINCTRL_ESP32_PINCFG_INIT(node_id)	\
-	(((ESP32_NO_PULL * DT_PROP(node_id, bias_disable)) << ESP32_PIN_BIAS_SHIFT) |		\
-	 ((ESP32_PULL_UP * DT_PROP(node_id, bias_pull_up)) << ESP32_PIN_BIAS_SHIFT) |		\
-	 ((ESP32_PULL_DOWN * DT_PROP(node_id, bias_pull_down)) << ESP32_PIN_BIAS_SHIFT) |	\
-	 ((ESP32_PUSH_PULL * DT_PROP(node_id, drive_push_pull)) << ESP32_PIN_DRV_SHIFT) |	\
-	 ((ESP32_OPEN_DRAIN * DT_PROP(node_id, drive_open_drain)) << ESP32_PIN_DRV_SHIFT) |	\
-	 ((ESP32_PIN_OUT_HIGH * DT_PROP(node_id, output_high)) << ESP32_PIN_OUT_SHIFT) |	\
-	 ((ESP32_PIN_OUT_LOW * DT_PROP(node_id, output_low)) << ESP32_PIN_OUT_SHIFT))
+#define Z_PINCTRL_ESP32_PINCFG_INIT(group_id)							\
+	(((ESP32_NO_PULL * DT_PROP(group_id, bias_disable)) << ESP32_PIN_BIAS_SHIFT) |		\
+	 ((ESP32_PULL_UP * DT_PROP(group_id, bias_pull_up)) << ESP32_PIN_BIAS_SHIFT) |		\
+	 ((ESP32_PULL_DOWN * DT_PROP(group_id, bias_pull_down)) << ESP32_PIN_BIAS_SHIFT) |	\
+	 ((ESP32_PUSH_PULL * DT_PROP(group_id, drive_push_pull)) << ESP32_PIN_DRV_SHIFT) |	\
+	 ((ESP32_OPEN_DRAIN * DT_PROP(group_id, drive_open_drain)) << ESP32_PIN_DRV_SHIFT) |	\
+	 ((ESP32_PIN_OUT_HIGH * DT_PROP(group_id, output_high)) << ESP32_PIN_OUT_SHIFT) |	\
+	 ((ESP32_PIN_OUT_LOW * DT_PROP(group_id, output_low)) << ESP32_PIN_OUT_SHIFT))
 
 /**
  * @brief Utility macro to initialize each pin.
  *
- * @param node_id Node identifier.
- * @param state_prop State property name.
- * @param idx State property entry index.
+ * @param group_id State group identifier.
+ * @param prop State group property name.
```

Comment:
I think group_id is confusing, you can keep `node_id`, and mention that it corresponds to the group node identifier instead.

Reply:
ditto